ui/ozone/platform/wayland/wayland_data_offer.h

```diff
@@ -42,20 +42,36 @@ class WaylandDataOffer {
   // list of provided mime types so that Chrome clipboard's machinery
   // works fine.
   void EnsureTextMimeTypeIfNeeded();
+  void SetAction(uint32_t dnd_actions, uint32_t preferred_action);
+  void Accept(uint32_t serial, const std::string& mime_type);
+  void Reject(uint32_t serial);
 
   // Creates a pipe (read & write FDs), passing the write-end of to pipe
   // to the compositor (via wl_data_offer_receive) and returning the
   // read-end to the pipe.
   base::ScopedFD Receive(const std::string& mime_type);
+  void FinishOffer();
+  uint32_t source_actions();
+  uint32_t dnd_action();
 
  private:
   // wl_data_offer_listener callbacks.
   static void OnOffer(void* data,
                       wl_data_offer* data_offer,
                       const char* mime_type);
+  // Notifies the source-side available actions
+  static void OnSourceAction(void* data,
+                             wl_data_offer* offer,
+                             uint32_t source_actions);
+  // Notifies the selected action
+  static void OnAction(void* data, wl_data_offer* offer, uint32_t dnd_action);
 
   wl::Object<wl_data_offer> data_offer_;
   std::vector<std::string> mime_types_;
+  // actions offered by the data source
```

Comment:
Actions, Action

Reply:
It's a bitfield variable and could have multiple actions. https://people.freedesktop.org/~whot/wayland-doxygen/wayland/Client/structwl__data__offer__listener.html#ab0d4fe2ca5491f552bb8f953427eb763

Reply:
I meant that you should do s/actions/Actions/ and s/action/Action/